addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/discovery/SqueezeBoxPlayerDiscoveryService.java

```diff
@@ -0,0 +1,174 @@
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.squeezebox.handler.SqueezeBoxPlayer;
+import org.openhab.binding.squeezebox.handler.SqueezeBoxPlayerEventListener;
+import org.openhab.binding.squeezebox.handler.SqueezeBoxPlayerHandler;
+import org.openhab.binding.squeezebox.handler.SqueezeBoxServerHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * When a {@link SqueezeBoxServerHandler} finds a new SqueezeBox Player we will
+ * add it to the system.
+ * 
+ * @author Dan Cunningham
+ *
+ */
+public class SqueezeBoxPlayerDiscoveryService extends AbstractDiscoveryService
+		implements SqueezeBoxPlayerEventListener {
+	private final Logger logger = LoggerFactory
+			.getLogger(SqueezeBoxPlayerDiscoveryService.class);
+
+	private final static int TIMEOUT = 60;
+
+	private SqueezeBoxServerHandler squeezeBoxServerHandler;
+	private ScheduledFuture<?> requestPlayerJob;
+
+	/**
+	 * Discovers SqueezeBox Players attached to a SqueezeBox Server
+	 * 
+	 * @param squeezeBoxServerHandler
+	 */
+	public SqueezeBoxPlayerDiscoveryService(
+			SqueezeBoxServerHandler squeezeBoxServerHandler) {
+		super(SqueezeBoxPlayerHandler.SUPPORTED_THING_TYPES_UIDS, TIMEOUT, true);
+		this.squeezeBoxServerHandler = squeezeBoxServerHandler;
+		setupRequestPlayerJob();
+	}
+
+	@Override
+	protected void startScan() {
+		this.squeezeBoxServerHandler.requestPlayers();
+	}
+
+	@Override
+	protected void startBackgroundDiscovery() {
+		this.squeezeBoxServerHandler.requestPlayers();
+	};
+
+	@Override
+	protected void deactivate() {
+		super.deactivate();
+		if (requestPlayerJob != null) {
+			requestPlayerJob.cancel(true);
+			requestPlayerJob = null;
+		}
+	}
+
+	@Override
+	public void playerAdded(SqueezeBoxPlayer player) {
+		logger.debug("Player added {} {} ", player.getMacAddress(),
+				player.getName());
+		ThingUID bridgeUID = squeezeBoxServerHandler.getThing().getUID();
+		ThingUID thingUID = new ThingUID(SQUEEZEBOXPLAYER_THING_TYPE,
+				bridgeUID, player.getMacAddress().replace(":", ""));
+		Map<String, Object> properties = new HashMap<>(1);
+		properties.put("mac", player.getMacAddress());
+		DiscoveryResult discoveryResult = DiscoveryResultBuilder
+				.create(thingUID).withProperties(properties)
+				.withBridge(bridgeUID)
+				.withLabel("SqueezeBox Player " + player.getName()).build();
+		thingDiscovered(discoveryResult);
+	}
+
+	/**
+	 * Tells the bridge to request a list of players
+	 */
+	private void setupRequestPlayerJob() {
+		Runnable runnable = new Runnable() {
+			public void run() {
+				squeezeBoxServerHandler.requestPlayers();
```

Comment:
Here you can probably clean up older results (removeOlderResults) again? Just make sure that you do not accidentially remove results from other SqueezeBox servers as well (as you can only delete specific thing type results, but not based on ThingUIDs).